rcgen/src/key_pair.rs

```diff
@@ -689,7 +773,7 @@ impl<T> ExternalError<T> for Result<T, pem::PemError> {
 	}
 }
 
-pub(crate) trait PublicKeyData {
+pub trait PublicKeyData {
```

Comment:
Needs some tweaks, as discussed here, please put those in a separate commit.

https://github.com/rustls/rcgen/issues/282#issuecomment-2206239673

Also needs documentation.